compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -716,12 +716,16 @@ object SymDenotations {
      *  TODO: Find a more robust way to characterize self symbols, maybe by
      *       spending a Flag on them?
```

Comment:
I believe a flag *has* been spent on that since this was written, namely `SelfName`. Should this whole method become
```scala
    final def isSelfSym(using Context): Boolean = is(SelfName)
```
?

Reply:
https://github.com/lampepfl/dotty/pull/19073